app/forms/user_form.rb

```diff
@@ -10,8 +10,6 @@ class UserForm
 
   set_form_variables :user_code, current_user: :find_current_user
 
-  validate :check_user
```

Comment:
issue with this is the 'valid?' check was getting called after the user had been created/updated so if there was a problem with the user who was doing the editing not having correct permissions it would error but not roll it back

Reply:
Yep, in the form should be free of security, use should be valid by the time it reaches the form. So the check_user should be handled in the controller

Reply:
A first step could be to create the form in a before_action in the controller, then call check_user on the form still
Form should deal with contents of form, not security. Thats the controllers job
This would help with the rollbacks too